pkg/ruler/compat.go

```diff
@@ -295,7 +295,7 @@ func DefaultTenantManagerFactory(
 			GroupEvaluationContextFunc: FederatedGroupContextFunc,
 			ExternalURL:                cfg.ExternalURL.URL,
 			NotifyFunc:                 rules.SendAlerts(notifier, cfg.ExternalURL.String()),
-			Logger:                     log.With(logger, "user", userID),
+			Logger:                     log.With(logger, "component", "ruler", "insight", true, "user", userID),
```

Comment:
Open question is how specific do we want to get with this "component" key.

The logger here is not for the full ruler, just the prometheus Rule Manager, so we could potentially be more specific if we wanted to demarcate the difference.

Reply:
What's the use case for the `component` field? IIRC so far, in Mimir, we've used `component` for high level component, so from this perspective `component=ruler` makes sense, but I guess it really depends what you want to use it for.

Reply:
The use case I would *think* is to have a simpler label to filter & aggregate on rather than the ones we currently have which have to be aggregated across k8s namespaces - for example tracking down these logs we used the query `{job=~"(cortex|mimir).*/ruler"}`

There is some inconsistency in Mimir as to how high-level or low-level the component label is.

Some places are pretty high-level, there are usages of component that are just `ruler`, `compactor`, and `query-frontend`.

But within the Alertmanager code in particular, the use of component is more specific: `nflog`, `silences`, `api`, `dispatcher`, and `MultiTenantAlertmanager`.

I think we stick with `"component": "ruler"` here because we already have one usage of "ruler" as a  component label within `RecordAndReportRuleQueryMetrics`, and because it matches the general usage pattern we see if we exclude the Alertmanager examples.